cfme/middleware/provider/__init__.py

```diff
@@ -82,6 +87,80 @@ class MiddlewareProvider(BaseProvider):
     add_provider_button = form_buttons.FormButton("Add this Middleware Provider")
     save_button = form_buttons.FormButton("Save Changes")
     taggable_type = 'ExtManagementSystem'
+    appliance = CurrentAppliance()
+
+
+@navigator.register(MiddlewareProvider, 'All')
+class All(CFMENavigateStep):
+    prerequisite = NavigateToAttribute('appliance', 'LoggedIn')
+
+    def step(self):
+        from cfme.web_ui.menu import nav
+        nav._nav_to_fn('Middleware', 'Providers')(None)
+
+    def resetter(self):
+        # Reset view and selection
+        tb.select("Grid View")
+        sel.check(paginator.check_all())
+        sel.uncheck(paginator.check_all())
+
+
+@navigator.register(MiddlewareProvider, 'Add')
+class Add(CFMENavigateStep):
+    prerequisite = NavigateToSibling('All')
+
+    def step(self):
+        cfg_btn('Add a New Middleware Provider')
+
```

Comment:
Not seeing middleware on my appliance so I can't answer my own question - is 'Discover' a valid destination here, even though it may not be used in existing tests?